src/main/java/org/dependencytrack/resources/v1/DependencyGraphResource.java

```diff
@@ -0,0 +1,114 @@
+
+    @GET
+    @Path("/search/{uuid}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            value = "Returns a list of specific components and services from project or component UUID",
+            response = DependencyGraphResponse.class
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 401, message = "Unauthorized"),
+            @ApiResponse(code = 403, message = "Access to a specified component is forbidden"),
+            @ApiResponse(code = 404, message = "Any component can be found"),
+    })
+    @PermissionRequired(Permissions.Constants.VIEW_PORTFOLIO)
+    public Response getComponentsAndServicesByParentUuid(@PathParam("uuid") String uuid) {
+        try (QueryManager qm = new QueryManager()) {
+            String directDependenciesJSON;
+            final Project project = qm.getObjectByUuid(Project.class, uuid);
+
+            if (project != null) {
+                if (qm.hasAccess(super.getPrincipal(), project)) {
+                    directDependenciesJSON = project.getDirectDependencies();
+                } else {
+                    return Response.status(Response.Status.FORBIDDEN).entity("Access to the specified project is forbidden").build();
+                }
+            } else {
+                final Component component = qm.getObjectByUuid(Component.class, uuid);
+                if (component != null) {
+                    if (qm.hasAccess(super.getPrincipal(), component.getProject())) {
+                        directDependenciesJSON = component.getDirectDependencies();
+                    } else {
+                        return Response.status(Response.Status.FORBIDDEN).entity("Access to the specified component is forbidden").build();
+                    }
+                } else {
+                    return Response.status(Response.Status.NOT_FOUND).entity("The component could not be found.").build();
+                }
+            }
+
+            if (directDependenciesJSON != null) {
+                JsonArray directDependencies = Json.createReader(new StringReader(directDependenciesJSON)).readArray();
+                final List<Component> components = new ArrayList<>(directDependencies.size());
+                final List<ServiceComponent> services = new ArrayList<>(directDependencies.size());
+                final List<UUID> uuids = directDependencies.stream().map(directDependency -> directDependency.asJsonObject().getString("uuid")).map(UUID::fromString).toList();
+
+                components.addAll(qm.getObjectsByUuids(Component.class, uuids));
+                services.addAll(qm.getObjectsByUuids(ServiceComponent.class, uuids));
+
+                return Response.ok(new DependencyGraphResponse(components, services)).build();
+            } else {
+                return Response.ok(new DependencyGraphResponse(new ArrayList<>(), new ArrayList<>())).build();
+            }
+        }
```

Comment:
@sonatype-lift ignore

The resource `ProjectQueryManager` is not released, but it doesn't have to.
The only critical resource of `ProjectQueryManager` is the `PersistenceManager` which is the same instance as `QueryManager`. So when `QueryManager` is close, all critical resources of `ProjectQueryManager` are closed.
The `QueryManager` is close at the end of the `try` scope because it's a `try-with-resources`.